presentation/diarywrite/src/main/java/com/hilingual/presentation/diarywrite/DiaryWriteViewModel.kt

```diff
@@ -75,4 +78,22 @@ internal class DiaryWriteViewModel @Inject constructor(
             }
         }
     }
+
+    fun extractTextFromImage(uri: Uri) {
+        viewModelScope.launch {
+            try {
+                val extractedText = textRecognitionManager.recognizeTextFromImage(uri)
+
+                val filteredText = if (extractedText.length > 1000) {
+                    extractedText.substring(0, 1000)
+                } else {
+                    extractedText
+                }
```

Comment:
p3
```suggestion
                val filteredText = extractedText.run {
        if (length > 1000) this.take(1000) else this
    }
```
이것처럼 `substring` 말고도 `take`를 사용하는 방법도 있어요~!